### ISSUE TITLE
Bun support

### ISSUE THREAD

**Issue by dmastrorillo** [joshuajaco/eslint-plugin-workspaces#45]
### Description
I'm using `eslint-plugin-workspaces` in a Turborepo monorepo managed by Bun (v1.3.1), and the `workspaces/require-dependency` rule doesn't appear to be detecting when workspace packages are imported without being declared in the consuming package's `dependencies` or `devDependencies`.

I noticed that Bun is not explicitly listed as a supported package manager in the README (which mentions npm, yarn, and pnpm). I'd like to request support for Bun workspaces, or at least understand if there are any plans to support it in the future.

### Current Behavior
When I import a workspace package that is NOT declared in my `package.json` dependencies, the `workspaces/require-dependency` rule does not flag it as an error.

**Example:**
```typescript
// tools/cli/src/commands/workspace/ls.ts
import { getCurrentPackages } from "@simply-dev/utils/get-current-packages";
```

Package that's importing `(tools/cli/package.json)`:
```json
{
  "name": "@simply-dev/cli",
  "dependencies": {
    "@oclif/core": "^4.8.0",
    "oclif": "^4.22.59",
    "plop": "catalog:"
  },
  "devDependencies": {
    "@simply-dev/typescript-config": "workspace:*",
    "@simply-dev/eslint-config": "workspace:*"
  }
}
```
Note that `@simply-dev/utils` is not declared in dependencies or devDependencies, but the import works due to Bun's workspace resolution. The `workspaces/require-dependency` rule should flag this but doesn't.

Package being imported `(tools/utils/package.json)`:
```json
{
  "name": "@simply-dev/utils",
  "exports": {
    "./get-current-packages": "./src/get-current-packages.ts",
    "./get-workspaces": "./src/get-workspaces.ts",
    "./get-root-dir": "./src/get-root-dir.ts"
  }
}
```

## Expected Behaviour

The `workspaces/require-dependency` rule should detect that @simply-dev/utils is being imported but not declared as a dependency, and flag it as an error.

## ESLint Configuration

Using ESLint flat config with the recommended preset:

```js
import workspaces from "eslint-plugin-workspaces";

const defaultConfig = defineConfig(
  // ... other configs
  workspaces.configs["flat/recommended"],
  {
    // ... other rules
  }
);
```

## Root package.json
```json
{
  "name": "thesimplyinitiative",
  "private": true,
  "packageManager": "bun@1.3.1",
  "workspaces": [
    "packages/*",
    "apps/*",
    "configs/*",
    "tools/*"
  ]
}
```

**Comment by joshuajaco** [comment]
Thanks for the detailed report!

Although Bun isn’t explicitly listed as supported, the plugin does read the standard `workspaces` field in `package.json`.
Since Bun uses the same convention, it should work out of the box.

I put together a small repro using Bun to double-check this: https://github.com/joshuajaco/bun-monorepo-test

When running `bun eslint .`, I get the expected error:

```
/tools/cli/src/commands/workspace/ls.ts
  1:1  error  Importing from another workspace without listing it as a dependency is not allowed  workspaces/require-dep
```

One possible difference is that your ESLint flat config may not be applying the plugin to TypeScript files.
Depending on how your config is structured, you may need to explicitly specify the file patterns, for example:

```js
defineConfig(
  // ...
  {
    files: ['**/*.ts'],
    ...workspaces.configs['flat/recommended'],
  }
);
```

If that doesn’t resolve it, feel free to share a minimal repro and I can take another look.

**Comment by dmastrorillo** [comment]
Sorry mate only just saw your comment, I'll check it out soon and get back to you!

**Comment by joshuajaco** [comment]
Thanks for the reproduction! Should be fixed in `v0.12.1`.

**Comment by dmastrorillo** [comment]
Confirmed working @joshuajaco thanks !
